magento/magento2
# [WIP] Refresh customer data when configurable product has been deleted

### Description (*)
This pull request fixes an issue with mini cart not being refreshed when a configurable product has been deleted from the website. It affects **only** active browser sessions where customers added the deleted product to the cart. That happens because the deletion of a product from admin (API) does not trigger the refresh of customer browser local storage. The non-existent product gets deleted from the cart when the session expires or when a customer interacts with the checkout session, e.g. adds a new product to the cart (that triggers quote and customer data refresh). 

### Manual testing scenarios (*)

1. Add a configurable product to the cart on the storefront
2. Go to the shopping cart
3. Login to the admin in a new browser tab
4. Delete a **configurable** product added to the cart on the first step (it's important to delete a configurable product because deletion of a simple product which represents a configurable option has a different effect)
5. Return to the storefront browser tab and refresh the page.
6. Actual result - shopping cart is empty but the mini cart is not empty (it contains deleted product)
![Empty_cart_bug](https://user-images.githubusercontent.com/12259690/87807306-c96cfc80-c860-11ea-81f5-bedd37cbac55.png)
Expected result - shopping cart and mini-cart should be empty.

### Questions or comments

This fix has a limitation because it requires a request from the front end to hit the Magento backend. It will work fine for non-cacheable pages or pages which are not in the cache yet or when there's an ajax request sent from the page.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#30039: [WIP] Refresh customer data when configurable product has been deleted
